server/punishments.ts

```diff
@@ -172,13 +172,14 @@ export const Punishments = new class {
 		['BATTLEBAN', 'battlebanned'],
 		['MUTE', 'muted'],
 	]);
-	constructor() {
-		setImmediate(() => {
-			void Punishments.loadPunishments();
-			void Punishments.loadRoomPunishments();
-			void Punishments.loadBanlist();
-			void Punishments.loadSharedIps();
-		});
+
+	load() {
```

Comment:
I still don't like the requirement that you need to call `load` after requiring this file, or the file will crash. I really don't like sacrificing ergonomics for some random purity requirement that has zero practical benefit.

Calling `load` at the bottom of this file will give nearly the same benefits, anyway. I'm pretty sure.

Reply:
> I still don't like the requirement that you need to call `load` after requiring this file, or the file will crash. I really don't like sacrificing ergonomics for some random purity requirement that has zero practical benefit.

Needing to call `load` after importing definitely isn't useless, it's nice to be able to be sure that state was loaded before using `Punishments.userids` or whatever.

> Calling `load` at the bottom of this file will give nearly the same benefits, anyway. I'm pretty sure.

It's currently ran in `index.ts`, which is where PS itself imports Punishments. I'm not sure if it's a good idea to make synchronous loading an inherent part of Punishments.

Reply:
Hm, yeah, I can see the merits of that approach. I suppose I'll leave it up to you.

Reply:
I'll choose to keep the current status quo of this PR.